ksql-api/src/main/java/io/confluent/ksql/api/server/PushQueryId.java

```diff
@@ -16,24 +16,21 @@
 package io.confluent.ksql.api.server;
 
 import java.util.Objects;
-import java.util.UUID;
 
 /**
- * Handle to a query that is passed to the client on query creation and can subsequently be used to
- * close a query. Uses UUID.randomUUID() which internally uses SecureRandom - this makes the id
- * cryptographically secure. This is important as we don't want random users guessing query IDs and
- * closing other peoples queries.
+ * Surprisingly, the id of a push query
  */
-public final class ApiQueryID {
+public class PushQueryId {
 
   private final String id;
 
-  public ApiQueryID() {
-    this.id = UUID.randomUUID().toString();
+  public PushQueryId(final String id) {
+    this.id = id;
```

Comment:
nit: validate params that will be stored in object state; ensuring object does not get into an invalid state.

Reply:
ack